crest/Assets/Crest/Crest/Scripts/Shapes/FFT/ShapeFFT.cs

```diff
@@ -209,7 +203,19 @@ public void CrestUpdate(CommandBuffer buf)
             // If geometry is being used, the ocean input shader will rotate the waves to align to geo
             var windDirRad = _meshForDrawingWaves != null ? 0f : _waveDirectionHeadingAngle * Mathf.Deg2Rad;
             var windSpeedMPS = (_overrideGlobalWindSpeed ? _windSpeed : OceanRenderer.Instance._globalWindSpeed) / 3.6f;
-            var waveData = _compute.GenerateDisplacements(buf, _resolution, _windTurbulence, windSpeedMPS, windDirRad, OceanRenderer.Instance.CurrentTime, _activeSpectrum, updateDataEachFrame);
+
+            // Don't create tons of generators when values are varying. Notify so that existing generators may be adapted.
+            if (_windTurbulenceOld != _windTurbulence || _windDirRadOld != windDirRad || _windSpeedOld != windSpeedMPS || _spectrumOld != _spectrum)
+            {
+                FFTCompute.OnGenerationDataUpdated(_resolution, _windTurbulenceOld, _windDirRadOld, _windSpeedOld, _spectrumOld, _windTurbulence, windDirRad, windSpeedMPS, _spectrum);
+            }
```

Comment:
Yeah it's a slightly odd pattern. If multiple wave components share one FFT, then one of them changes its settings, it will actually steal the generator from the rest. Then the first from the rest which request the old settings will trigger creation of a new generator, and the remaining ones will use this new generator. In the end one new generator is created, but it's created for the old settings. Generators are requested single threaded so there should not be a race condition. Odd pattern but I don't think any other way works without ugly checks to see if old generators are still used, or other complicated things.

I will add a comment to this effect to the code.